### ISSUE TITLE
ToggleButtonIO conversion to TypeScript

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/tandem#259]
I attempted to convert sun.ToggleButtonIO to TypeScript, and ran into several problems/questions.

(1) RectangularToggleButton et. al. require parameterization (e.g. RectangularToggleButton<boolean>) which is not supported by IOType.

(3) Why is there a 'phet.sun.' prefix on these values? Why aren't imports used?

```js
const types = [ phet.sun.RectangularToggleButton, phet.sun.RoundStickyToggleButton, phet.sun.RoundToggleButton ];
```

(2) What should the type be for `isValidValue: instance`?  Are we going to need this awful duplication?:

```js
  isValidValue: ( instance: RectangularToggleButton |  RoundStickyToggleButton  | RoundToggleButton  ) => {
    const types = [ RectangularToggleButton, RoundStickyToggleButton, RoundToggleButton ];
```

Assigning to @samreid and @zepumph.  FYI @jbphet.

**Comment by samreid** [comment]
I observed that the main reason for ToggleButtonIO existing is that it defines an event type `'toggled'`.  However, that is an old pattern that predates using `Emitter`s for the same thing.  I observed that there are no `'toggled'` events in the codebase at the moment.  Therefore, it seems `ToggleButtonIO` can be removed.

**Comment by samreid** [comment]
I removed `ToggleButtonIO` in the commits, @zepumph can you please review at your convenience?

**Comment by zepumph** [comment]
Yes excellent. I can confirm that we have a toggledEmitter instrumented in the button model. This is the right strategy. Thanks!

https://github.com/phetsims/sun/blob/b30e57184a4740d4a9501968924234a86bb6f6f2/js/buttons/ToggleButtonModel.ts#L59-L63

**Comment by samreid** [comment]
Thanks, it seems this issue can be closed.
